### DISCUSSION TITLE
optionValue support for p-treeSelect

### DISCUSSION THREAD

**Discussion by puschie286** [primefaces/community#2420] [PrimeNG]
Would be nice to have an optionValue property (like in all other select components).

Currently, selection gives you a TreeNode or TreeNode[]. This makes it pretty complicated to be used within a reactive form.

The only solution i see right now is to create transform method for each treeSelect (and mark element as standalone with ngModelOptions )

**Comment by puschie286** [comment]
helper function:
```ts
export function getNodeFromKey( key?: string, nodes?: TreeNode[] ): TreeNode | undefined
{
	if( key == undefined || nodes == undefined || nodes.length === 0 )
	{
		return undefined;
	}

	for( const entry of nodes )
	{
		// hit ?
		if( entry.key === key )
		{
			return entry;
		}

		// level down
		const found = getNodeFromKey( key, entry.children );
		if( found != undefined )
		{
			return found;
		}
	}

	return undefined;
}
```

**Comment by RobersonLuo** [comment]
Yes！
1、It's to complicated to use TreeNode as the 'tree-select' component's data.  I think use ordinary object is ok (not TreeNode object).
2、No 'optionValue' property to binding to the DTO property (this function is too important when used within reactive form).
